src/core/config/config_module.rs

```diff
@@ -15,13 +15,22 @@ use crate::core::rest::{EndpointSet, Unchecked};
 
 /// A wrapper on top of Config that contains all the resolved extensions and
 /// computed values.
-#[derive(Clone, Debug, Default, Setters, MergeRight)]
+#[derive(Clone, Debug, Default, Getters, MergeRight)]
 pub struct ConfigModule {
-    pub config: Config,
-    pub extensions: Extensions,
-    pub input_types: HashSet<String>,
-    pub output_types: HashSet<String>,
-    pub interface_types: HashSet<String>,
+    config: Config,
+    extensions: Extensions,
+    input_types: HashSet<String>,
+    output_types: HashSet<String>,
+    interface_types: HashSet<String>,
+}
+
+impl ConfigModule {
+    pub fn config_mut(&mut self) -> &mut Config {
+        &mut self.config
+    }
```

Comment:
Drop this method, instead add a modify method

```rust
fn modify(f: Fn(Config) -> Config) -> Self {
  ...
}
```

The modify method will re-create all the internal fields such as input_types and output_types etc.